crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -56,13 +56,13 @@ use tracing::{debug, error, info, instrument, trace, warn};
 /// * [`BlockchainTree::make_canonical`]: Check if we have the hash of a block that is the current
 ///   canonical head and commit it to db.
 #[derive(Debug)]
-pub struct BlockchainTree<DB, E> {
+pub struct BlockchainTree<DB, C, E> {
     /// The state of the tree
     ///
     /// Tracks all the chains, the block indices, and the block buffer.
     state: TreeState,
     /// External components (the database, consensus engine etc.)
-    externals: TreeExternals<DB, E>,
+    externals: TreeExternals<DB, C, E>,
```

Comment:
the Consensus type is super invasive and would introduce generics everywhere, I'd like to keep using `Arc<dyn Consensus>`

Reply:
Fair enough - I wasn't sure whether the use of `dyn` was actually preferred or only used as a placeholder for a future generic.

Reply:
> developer overhead anywhere that this struct is utilized.

yeah we initial had this as a generic but pivoted to dyn because trait is object safe and the functions aren't called frequently.

we could likely use dyn for a few more generics but some aren't object safe, although we could likely make them, like Database